apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/ListOrders/CompletedOrders.tsx

```diff
@@ -41,40 +42,46 @@ export type MappedRawOrder = {
   date: string
 }
 
+const compareByDate = (a: MappedRawOrder, b: MappedRawOrder) => {
+  return new Date(b.date).getTime() - new Date(a.date).getTime()
+}
+
 const findCompletedOrderTx = (transactions: TransactionInfo[]): MappedRawOrder[] => {
   const sentTransactions = transactions.filter((tx) => tx.direction === 'SENT')
   const receivedTransactions = transactions.filter((tx) => tx.direction === 'RECEIVED')
 
-  const filteredTx = sentTransactions.reduce((acc, sentTx) => {
-    const result: {id?: string; metadata?: TxMetadataInfo; date?: string} = {}
-    receivedTransactions.forEach((receivedTx) => {
-      receivedTx.inputs.forEach((input) => {
-        if (Boolean(input.id) && input?.id?.slice(0, -1) === sentTx?.id) {
-          result['id'] = sentTx?.id
-          result['metadata'] = sentTx?.metadata
-          result['date'] = receivedTx?.lastUpdatedAt
-        }
+  const filteredTx = sentTransactions
+    .reduce((acc, sentTx) => {
+      const result: {id?: string; metadata?: TxMetadataInfo; date?: string} = {}
+      receivedTransactions.forEach((receivedTx) => {
+        receivedTx.inputs.forEach((input) => {
+          if (Boolean(input.id) && input?.id?.slice(0, -1) === sentTx?.id && receivedTx.metadata !== null) {
+            result['id'] = sentTx?.id
+            result['metadata'] = sentTx?.metadata
+            result['date'] = receivedTx?.lastUpdatedAt
+          }
+        })
       })
-    })
-
-    if (result['id'] !== undefined && result['metadata'] !== undefined) {
-      try {
-        const metadata = JSON.parse(result.metadata as string)
-        result['metadata'] = metadata
-        return acc.concat(result as MappedRawOrder)
-      } catch (error) {
-        console.warn('Error parsing json metadata', error)
-      }
-    }
-    return acc
-  }, [] as Array<MappedRawOrder>)
 
-  return filteredTx.filter((tx) => tx.metadata !== null)
+      if (result['id'] !== undefined && result['metadata'] !== undefined) {
+        try {
+          const metadata = JSON.parse(result.metadata as string)
+          result['metadata'] = metadata
+          return acc.concat(result as MappedRawOrder)
+        } catch (error) {
+          console.warn('Error parsing json metadata', error)
```

Comment:
would an alert be ok here? 

https://github.com/Emurgo/yoroi/blob/8068c57ba81cdbd960119fbac480f689e4f9edde/apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx#L81